lib/pacemaker/pcmk_fence.c

```diff
@@ -161,16 +161,20 @@ pcmk__fence_action(stonith_t *st, const char *target, const char *action,
 
     free(async_fence_data.name);
 
+    if ((reason != NULL) && (async_fence_data.result.exit_reason != NULL)) {
```

Comment:
we might still set *reason = NULL for the case caller doesn't initialize.

Reply:
yep